sodasql/scan/scan.py

```diff
@@ -112,7 +112,7 @@ def execute(self) -> ScanResult:
 
     def _query_columns_metadata(self):
         sql = self.warehouse.dialect.sql_columns_metadata_query(self.scan_yml.table_name)
```

Comment:
@vijaykiran I think we have a design decision to make here. The existing code assumes that there will be no need to do any transformation for getting the columns. 
This might not always be true. I believe that it should be the dialects responsibility to get the columns. The dialect can do anything that is needed to get the columns and convert the result to the coulmn tuples.

Reply:
Indeed, I can see Hive is a special case :) This would change the `Dialect`s responsibility a bit. For the first cut, I think this approach you have (below) seem okay to me.